### ISSUE TITLE
Intermittent failure of twisted.test.test_pb.NewStyleTestCase.test_newStyle

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3783]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3783|
|Type|defect|
|Created|2009-04-17 13:41:14Z|
|Branch|https://github.com/twisted/twisted/tree/disconnect-pb-servers-3783|

```
[ERROR]: twisted.test.test_pb.NewStyleTestCase.test_newStyle

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<Broker #0 on 51943>
```

Seen on buildbot, http://buildbot.twistedmatrix.com/builders/osx10.5-py2.5-select/builds/390

Perhaps related to [#2866](https://github.com/twisted/twisted/issues/2866)?  I dunno.

<details><summary>Searchable metadata</summary>

```
trac-id__3783 3783
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_disconnect_pb_servers_3783 branches/disconnect-pb-servers-3783
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1239975674000000 1239975674000000
changetime__1241055848000000 1241055848000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [26730]) Branching to 'disconnect-pb-servers-3783'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

I think this is done.  It's a bit annoying to verify because of [#3786](https://github.com/twisted/twisted/issues/3786) (at least when I try to run it - it can't get past the second run with trial -u).  However, with a fix for that ticket locally merged, I can run `twisted.test.test_pb.NewStyleTestCase` many thousands of times without failure with this branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Looks fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [26774]) Merge disconnect-pb-servers-3783

Author: exarkun
Reviewer: therve
Fixes: [#3783](https://github.com/twisted/twisted/issues/3783)

Change `twisted.test.test_pb.NewStyleTestCase` so that it tracks server-side
connections created during the course of tests and add a `tearDown` which
disconnects them all, fixing the intermittent unclean reactor errors from
`test_newStyle`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
